bitnami/kafka/templates/kafka-provisioning.yaml

```diff
@@ -122,9 +137,25 @@ spec:
             name: {{ include "kafka.log4j.configMapName" . }}
         {{ end }}
         {{- if (include "kafka.tlsEncryption" .) }}
-        - name: kafka-certificates
+        {{- if not (empty .Values.auth.tls.existingSecrets) }}
+        {{- range $index, $secret := .Values.auth.tls.existingSecrets }}
+        - name: kafka-certs-{{ $index }}
+          secret:
+            secretName: {{ tpl $secret $ }}
+            defaultMode: 256
+        {{- end }}
+        {{- else if or (not (empty (coalesce .Values.auth.tls.existingSecret .Values.auth.jksSecret))) (.Files.Glob "files/tls/*.{crt,pem}") }}
+        - name: kafka-certs
+          secret:
+            secretName: {{ if not (empty (coalesce .Values.auth.tls.existingSecret .Values.auth.jksSecret)) }}{{ tpl (coalesce .Values.auth.tls.existingSecret .Values.auth.jksSecret) . }}{{ else }}{{ include "kafka.fullname" . }}{{ end }}
+            defaultMode: 256
+        {{- else if .Values.auth.tls.autoGenerated }}
+        {{- range $index := until $replicaCount }}
+        - name: kafka-certs-{{ $index }}
           secret:
-            secretName: {{ include "kafka.tlsSecretName" . }}
+            secretName: {{ include "kafka.fullname" $ }}-{{ $index }}-tls
```

Comment:
I don't know if it is possible but maybe we can use the printf here?

{{ printf "%s-%d-tls" (include "common.names.fullname" .) ... }}

It is OK for me the way that you are creating the secretName value. Just a suggestion.